platform/viewer/src/components/StudyPrefetcher.js

```diff
@@ -1,29 +1,82 @@
-import { Component } from "react";
-import PropTypes from "prop-types";
-import OHIF from "@ohif/core";
+import React, { useEffect } from 'react';
+import { classes } from '@ohif/core';
+import PropTypes from 'prop-types';
+import cs from 'cornerstone-core';
+import { useSelector } from 'react-redux';
 
-const { StudyPrefetcher } = OHIF.classes;
+import './StudyPrefetcher.css';
 
-class StudyPrefetcherComponent extends Component {
-  static propTypes = {
-    studies: PropTypes.array
-  };
+const StudyPrefetcher = ({ studies, viewportIndex, options }) => {
+  const viewportData = useSelector(state => {
+    const { viewports = {} } = state;
+    const { activeViewportIndex, viewportSpecificData = {} } = viewports;
+    const viewportData = viewportSpecificData[activeViewportIndex] || {};
+    return viewportData;
+  });
 
-  componentDidMount() {
-    const { studies } = this.props;
+  const studyPrefetcher = classes.StudyPrefetcher.getInstance(studies, options);
 
-    this.studyPrefetcher = StudyPrefetcher.getInstance();
-    this.studyPrefetcher.setStudies(studies);
-  }
+  useEffect(() => {
+    studyPrefetcher.setStudies(studies);
 
-  render() {
-    return null;
-  }
+    const onImageRendered = ({ detail }) => {
+      console.debug('Prefetching...');
+      studyPrefetcher.prefetch(detail.element);
+      detail.element.removeEventListener(
+        cs.EVENTS.IMAGE_RENDERED,
+        onImageRendered
+      );
+    };
 
-  componentWillUnmount() {
-    // Stop prefetching when we close the viewer
-    this.studyPrefetcher.destroy();
-  }
-}
+    const activeEnabledElement = cs.getEnabledElements()[viewportIndex];
+    if (activeEnabledElement && activeEnabledElement.image) {
+      activeEnabledElement.element.addEventListener(
+        cs.EVENTS.IMAGE_RENDERED,
+        onImageRendered
+      );
+    }
```

Comment:
The prefetch will start only when the first image is rendered and the prefetch event will occur only once, that's why it's removed right after the first image, is rendered. In cornerstone, we don't have an Image rendered once event.

Reply:
This logic is self-descriptive I think, it's a once image rendered event handler. We don't need to run the same logic on every new image rendered for that viewport, otherwise, we would prefetch again on every new image scrolled.